coderd/database/migrations/000059_app_sharing_level.up.sql

```diff
@@ -0,0 +1,15 @@
+-- Add enum app_sharing_level
+CREATE TYPE app_sharing_level AS ENUM (
+	-- only the workspace owner can access the app
+	'owner',
+	-- the workspace owner and other users that can read the workspace template
+	-- can access the app
+	'template',
```

Comment:
I find this to be a bit of a strange access level. Typically when you think about sharing you think in terms of user abstractions (e.g. user, team, org, site) as opposed to something that is scoped to infrastructure. Also everyone in an org can read every template so there's no functional difference between `authenticated` and `template` in the AGPL version of Coder.

Removing this level will also simplify the code as you do not have to introduce the concept of templates when thinking about workspace connections.

Reply:
There's also maybe potential to support groups here in the future as a nice enterprise feature in lieu of the intention behind the templates level.

Reply:
Level template was suggested by Ammar. I agree it only really applies to enterprise so we should probably remove it for now. @ammario thoughts?

Reply:
Sounds good. We can add it back later.